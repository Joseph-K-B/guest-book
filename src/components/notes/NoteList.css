.notes {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    grid-template-rows: 1fr 1fr 1fr;
}

.note {
    display: flex;
    flex-direction: column;
    text-align: center;
    align-items: center;
}

.noteItem {
    position: relative;
    right: 1.25rem;
    min-width: 7.25rem;
    max-height: 7.25rem;
    display: flex;
    flex-direction: column;
    text-align: center;
    color: #F8F9FA;
    background-color: #607466;
    border: black solid 0.25rem;
    margin: 5%;
    box-shadow: 0.115rem 0.115rem black;
    border-radius: 10px;
}