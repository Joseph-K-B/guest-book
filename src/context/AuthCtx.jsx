import React, { createContext, useState } from "react"

const authCtx = createContext();

const AuthProvider = ({ children }) => {
  const [user, setUser] = useState(null);

  // This would probably be the sign up function 
  // b/c it includes both username and email
  // 
  const login = (username, email, password) => {
    const authConfirmed = (
      username === process.env.REACT_APP_AUTH_USER &&
      email === process.env.REACT_APP_AUTH_EMAIL &&
      password === process.env.REACT_APP_AUTH_PASSWORD
      )
      if(authConfirmed) 
        setUser({ username, email });
      return authConfirmed;
  };

  const logout = (cb) => {
    setUser(null);
    // cb();
  }

  return (
    <authCtx.Provider value={{user, setUser, login, logout}}>
      {children}
    </authCtx.Provider>
  );
};

export {authCtx, AuthProvider};