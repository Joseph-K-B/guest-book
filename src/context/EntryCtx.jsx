import { createContext, useContext, useState } from "react";

const NoteCtx = createContext();

const NoteProvider = ({children}) => {
    const [note, setNote] = useState('');

    return <NoteCtx.Provider value={{ note, setNote }}>{children}</NoteCtx.Provider>
}

const useNote = () => {
    const ctx = useContext(NoteCtx);

    if(ctx === undefined) {
        throw new Error('useNote hook can only be called within child components of NoteCtx Provider');
    }
    return ctx;
}

export { NoteProvider, useNote}