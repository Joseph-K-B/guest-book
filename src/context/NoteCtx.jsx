import React from 'react';
import { createContext, useContext, useState } from "react";

const NoteCtx = createContext();

const NoteProvider = ({ children }) => {
    const [notes, setNotes] = useState([]);

    return (
        <NoteCtx.Provider value={{ notes, setNotes }}>{children}</NoteCtx.Provider>
    )
}

const useNote = () => {
    const ctx = useContext(NoteCtx);

    if(ctx === undefined) {
        throw new Error('useNote hook can only be called within child components of NoteCtx Provider');
    }
    return ctx;
}

export { NoteProvider, useNote}