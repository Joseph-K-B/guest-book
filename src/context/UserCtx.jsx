import React from 'react';
import { createContext, useContext, useState } from "react"

const UserCtx = createContext();

const UserProvider = ({children}) => {
    const [name, setName] = useState('');

    return <UserCtx.Provider value={{ name, setName }}>{children}</UserCtx.Provider>
}

const useUser = () => {
    const ctx = useContext(UserCtx);

    if(ctx === undefined) {
        throw new Error('useUser custom hook must be called within child components of UserCtx Provider');
    }
    return ctx;
}

export { UserProvider, useUser }