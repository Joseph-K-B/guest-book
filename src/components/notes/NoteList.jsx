import { useNote } from "../../context/NoteCtx";
import Note from "./Note";

function NoteList() {
    const { notes } = useNote();

    return (
        <ul>
            {notes.map((note) => {
                return (
                    <li>
                        <Note note={note} />
                    </li>
                )
            })}
        </ul>
    )
}

export default NoteList;