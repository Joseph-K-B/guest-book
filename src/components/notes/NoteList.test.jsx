import React from 'react';
import { screen, render } from '@testing-library/react';
import { NoteProvider } from '../../context/NoteCtx';
import NoteList from './NoteList';

it('renders list of notes left by users', async () => {
    const {container} = render(
        <NoteProvider>
            <NoteList />
        </NoteProvider>
    );


    expect(container).toMatchSnapshot();
})