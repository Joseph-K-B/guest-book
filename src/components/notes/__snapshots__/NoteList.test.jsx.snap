// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list of notes left by users 1`] = `
<div>
  <ul />
</div>
`;
